### ISSUE TITLE
[Bug]: LC0095 should also be raised for local procedures

### ISSUE THREAD

**Issue by MODUSCarstenScholling** [ALCops/Analyzers#319]
### Analyzer

LinterCop

### Issue type

False negative (diagnostic doesn't fire when it should)

### Rule ID

LC0095

### Description

LC0095 should also be raised for local procedures. Below procedures with identical signature behave differently.

### AL code to reproduce

```al
    // Does not fire (wrong)
    local procedure InstallAppPerDatabase(appInfo: ModuleInfo; freshInstall: Boolean)
    begin
        // Use separate procedures for separate steps and do not directly code here
    end;

    // Fires (correct)
    internal procedure InstallAppPerCompany(appInfo: ModuleInfo; freshInstall: Boolean)
    begin
        // Use separate procedures for separate steps and do not directly code here
    end;
```

**Comment by Arthurvdv** [comment]
This behavior is currently considered to be by design, as local procedures are intentionally excluded from the LC0095 rule.

The reasoning behind this is documented here:

> ## Excluded from this rule
> Local procedures (AA0137 handles them)

https://alcops.dev/docs/analyzers/lintercop/lc0095/

In other words, LC0095 does not analyze local procedures because the responsibility for detecting this pattern was delegated to AA0137.

Looking at your example, I would therefore expect an AA0137 diagnostic to already be raised on the local procedure. Am I correct in assuming that this warning is already present in your case?

**Comment by MODUSCarstenScholling** [comment]
Oh, wow. I should have read the docs, but that felt too straightforward... Sorry!

No, I don't have AA0137. Wait. AA0137 is turned off in this install codeunit because it is a skeleton with some unused parameters.
So, you are right. Thank you for clarifying this, closing :)
